src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -145,9 +145,71 @@ void HSHomeObject::on_create_pg_message_commit(int64_t lsn, sisl::blob const& he
     if (ctx) ctx->promise_.setValue(folly::Unit());
 }
 
-PGManager::NullAsyncResult HSHomeObject::_replace_member(pg_id_t id, peer_id_t const& old_member,
-                                                         PGMember const& new_member) {
-    return folly::makeSemiFuture< PGManager::NullResult >(folly::makeUnexpected(PGError::UNSUPPORTED_OP));
+PGManager::NullAsyncResult HSHomeObject::_replace_member(pg_id_t pg_id, peer_id_t const& old_member_id,
+                                                         PGMember const& new_member, uint32_t commit_quorum) {
+
+    group_id_t group_id;
+    {
+        auto lg = std::shared_lock(_pg_lock);
+        auto iter = _pg_map.find(pg_id);
+        if (iter == _pg_map.end()) return folly::makeUnexpected(PGError::UNKNOWN_PG);
+        auto& repl_dev = pg_repl_dev(*iter->second);
+
+        if (!repl_dev.is_leader() && commit_quorum == 0) {
+            // Only leader can replace a member
+            return folly::makeUnexpected(PGError::NOT_LEADER);
+        }
+        group_id = repl_dev.group_id();
+    }
+
+    LOGI("PG replace member initated member_out={} member_in={}", boost::uuids::to_string(old_member_id),
+         boost::uuids::to_string(new_member.id));
+
+    homestore::replica_member_info in_replica, out_replica;
+    out_replica.id = old_member_id;
+    in_replica.id = new_member.id;
+    in_replica.priority = new_member.priority;
+    std::strncpy(in_replica.name, new_member.name.data(), in_replica.max_name_len);
```

Comment:
Can we enforce the `len(name) <=31` to `PGMember` constructor?  and better add a constructor for `homestore::replica_member_info`.

`strncpy` is risky here as it can truncate make the `in_replca.name` not ended with `\0`.  Later in the 
`pg->pg_info_.members.emplace(PGMember(member_in.id, member_in.name, member_in.priority));` we construct a string by a char array doesnt ended with `\0`, the behavior is risky.

Reply:
Made the max length to 127 and assert. Cant include homestore header files in pg_manager.hpp for maxlen.